mlflow/R/mlflow/R/model.R

```diff
@@ -111,88 +111,66 @@ mlflow_load_model <- function(model_uri, flavor = NULL, client = mlflow_client()
       warning(paste("Multiple model flavors available (", paste(available_flavors, collapse = ", "),
                     " ).  loading flavor '", available_flavors[[1]], "'", ""))
     }
-
     flavor <- available_flavors[[1]]
   }
-
   flavor_path <- model_path
   class(flavor_path) <- c(flavor, class(flavor_path))
   mlflow_load_flavor(flavor_path)
 }
 
-#' Predict using RFunc MLflow Model
-#'
-#' Performs prediction using an RFunc MLflow model from a file or data frame.
-#'
-#' @template roxlate-model-uri
-#' @param input_path Path to 'JSON' or 'CSV' file to be used for prediction.
-#' @param output_path 'JSON' or 'CSV' file where the prediction will be written to.
-#' @param data Data frame to be scored. This can be used for testing purposes and can only
-#'   be specified when `input_path` is not specified.
-#' @param restore Should \code{mlflow_restore_snapshot()} be called before serving?
-#'
-#' @examples
-#' \dontrun{
-#' library(mlflow)
-#'
-#' # save simple model which roundtrips data as prediction
-#' mlflow_save_model(function(df) df, "mlflow_roundtrip")
-#'
-#' # save data as json
-#' jsonlite::write_json(iris, "iris.json")
-#'
-#' # predict existing model from json data
-#' # load the model from local relative path.
-#' mlflow_rfunc_predict("file:mlflow_roundtrip", "iris.json")
-#' }
-#'
-#' @importFrom utils read.csv
-#' @importFrom utils write.csv
-#' @export
+# Generate predictions using a saved R MLflow model.
+# Input and output are read from and written to a specified input / output file or stdin / stdout.
+#
+# @param input_path Path to 'JSON' or 'CSV' file to be used for prediction. If not specified data is
+#                   read from the stdin.
+# @param output_path 'JSON' file where the prediction will be written to. If not specified,
+#                     data is written out to stdout.
+
+# @importFrom utils read.csv
```

Comment:
Made private. Since it is to be called by the mlflow model serve cli there is no need to accept R dataframe as input.

Reply:
Makes sense, given that the `mlflow_predict_model` API exists for predicting on dataframes. BTW as it stands we have:
`mlflow_predict_model` for predicting on an R dataframe using an R model
`mlflow model predict` CLI for predicting on data in a file

Would we eventually want to add an R wrapper for the `mlflow model` CLI & if so how would we name it to avoid collisions with `mlflow_predict_model`? Maybe `mlflow_model_cli_predict` and `mlflow_model_cli_serve` or something?

Reply:
We could just remove `mlflow_predict_model()` and for the CLI wrappers we'll call them `mlflow_model_predict()` and `mlflow_model_serve()`. The current `mlflow_predict_model()` isn't really necessary since the loaded models have their own predict methods.

Reply:
@kevinykuo  Do you mean that in order to score an mlflow model you would simply call predict(model, data) instead of mlflow_model_predict(model, data)?

Reply:
Currently there's no "mlflow model" object in R, i.e. the loaded models are just model objects defined by the package that created them. If a `predict()` method is implemented for those classes, you can call `predict()`, but e.g. for mleap the function is called `mleap_transform()`.

Reply:
I see. Then I think we need the mlflow_predict_model cause we want to have a unified interface.